seps/sep-0002.md

```diff
@@ -0,0 +1,169 @@
+
+With this approach, can we efficiently determine which group members should receive a notification when a post is created on that group's page?  How challenging would it be to ensure that only group members can post to this group's page? There is a solution here, however these implicitly defined group connections with multiple versions of group membership adds complexity. 
+
+
+1. [Communities on DevHub](https://near.org/devgovgigs.near/widget/Ideas)
+
+DevHub's community implementation exists on a distinct contract from near.social. This results in a lack of discoverability of DevHub's groups across gateways due to its community data not being included in the indexes which power search on socialdB.
+
+
+## Specification 
+Building upon the NFT-based groups implementation, we would benefit from the following the creation and maintanence of two [Core Social Indexers](https://docs.google.com/document/d/14OmYzMYX60lBm2U2Ridz6GZCeN_A6IcU_qn-yPwLTfs/edit#heading=h.nfrlp9tpavvv):
+
+1. A dataplatform.near Core Social indexer for NFT_Groups, that tracks posts to main which include a groupContractId, e.g. writes to post.main.[groupContractId], and provides an API which lists all of the groups which have been created in socialDB. 
+	1. To support the ability to search for groups, where search text is matched against the group's metadata such as name, symbol, description, and spec, the above indexer should fetch and store this metadata from a source such as the [Mintbase API](https://cloud.hasura.io/public/graphiql?endpoint=https%3A%2F%2Fgraph.mintbase.xyz%2Fmainnet&header=mb-api-key%3Aanon&query=query+MyQuery+%7B%0A++nft_metadata%28%0A++++where%3A+%7Breference%3A+%7B_eq%3A+%22nb0-oBR379DzoFYeYv-LesjVNmrVlDs5IqQ8hfDfnMU%22%7D%7D%0A++%29+%7B%0A++++id%0A++++media%0A++++reference_blob%0A++%7D%0A%7D%0A
+
+1. A dataplatform.near Core Social indexer for Group_Members, that maintains a list of NFT owners for each NFT_Group, and provides an API which serves a list of all AccountIds which belong to a group.
+
+
+## Drawbacks
+
+- The NFT-based groups implementation shifts the responsibility of valid NFT management onto users. However, gateways can mitigate this with a Mintbase integration or the ContractWizardUI to faciliate the creation of NFTs.
```

Comment:
I have a question: is there a good practice to push this common logic to all gateways, including near.org/near.social and 3rd party gateways?

Reply:
I think we may want to advocate that gateways use a specific component that facilitates the creation of an NFT, we could start with [this one](https://near.org/contractwizard.near/widget/ContractWizardUI)

Reply:
Got it. Thanks!